### PR TITLE
pika-algorithms: new port in devel

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

New port

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6.8
Xcode 3.2.6

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: devel/pika-algorithms/Portfile [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+platforms           {darwin any}
+license             Boost-1
+maintainers         {@barracuda156 gmail.com:vital.had} openmaintainer
+description         ${name} is a C++ library containing parallel algorithms built on top of pika
+long_description    {*}${description}
+checksums           rmd160  1063dd5aac28ff6d3e803205a80324abe1901b8a \
+                    sha256  67ea5e8545b234f82dcc75612a774f2e3df8425a283f2034c2d1e2e5ac74f945 \
+                    size    832806
+github.tarball_from archive
+supported_archs     noarch
```
Comment (reneeotten) [inline devel/pika-algorithms/Portfile:24]: why would this be noarch - there are clearly things being compiled...

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: devel/pika-algorithms/Portfile (hunk 1 above)
Comment (barracuda156) [reply devel/pika-algorithms/Portfile:24]: It installs headers and CMake configuration files, no binaries or libs.
I should perhaps check if anything arch-specific is getting written down into CMake files though in a text form.

Comment (reneeotten) [reply devel/pika-algorithms/Portfile:24]: why is there all the mpi stuff and compiler selection going in if nothing is being build. That seems all very counter intuitive.

Comment (barracuda156) [reply devel/pika-algorithms/Portfile:24]: I made the settings to match those of Pika itself (which does use MPICH etc), and configure of this port checks for all that stuff.
It also does compile things if tests are built.

**Comment by barracuda156** [comment]
@reneeotten No point in merging this, I am not gonna carry a dedicated version of `pika` for compatibility, and upstream dropped support for `pika-algorithms` in 0.21.0: https://github.com/pika-org/pika-algorithms/issues/24#issuecomment-1859836856

**Pull request closed without merging**